### ISSUE TITLE
Add a summary of how it works in the readme

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/autodoc-traits#11]
I think having a "How it works" section ([example from jupyterhub-idle-culler](https://github.com/jupyterhub/jupyterhub-idle-culler#how-it-works)) in the readme would be great. I'm currently debugging an issue where I look to determine if its an issue with autodoc_traits, sphinx.ext.autodoc, myst_parser, or something else. Not understanding how all the pieces work makes me unable to figure out what my problem is.

## Current understanding of autodoc_traits

- autodoc_traits registeres two new "autodocumenter" classes that sphinx.ext.autodoc will honor using [`add_autodocumenter`](https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.add_autodocumenter)
  https://github.com/jupyterhub/autodoc-traits/blob/d618d9e57023b00575436a0928c231e7e779bfe3/autodoc_traits/autodoc_traits.py#L60-L62
- autodoc_trait's autodocumenter classes include details like below, which helps the sphinx.ext.autodoc directives decide if and when the autodocumenter is relevent. The best documentation of these I've found are in [this example](https://www.sphinx-doc.org/en/master/development/tutorials/autodoc_ext.html#autodoc-ext-tutorial)
  https://github.com/jupyterhub/autodoc-traits/blob/d618d9e57023b00575436a0928c231e7e779bfe3/autodoc_traits/autodoc_traits.py#L36-L40
- From here on, autodoc_traits work is done, and sphinx.ext.autodoc takes over using the added autodocumenter classes when suitable
- sphinx.ext.autodoc will generate [`.. py:` directives](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#the-python-domain) with provided docstrings etc.

To summarize, `autodoc_traits` will register autodocumenter classes with `sphinx.ext.autodoc`, which in turn will scan source code and emit rST declarations of `.. py:` directives.

## Request for review

I can document this in the README, but I'd love some input/review of this understanding. Review questions could be:
- is it accurate?
- does it capture how things work?
- is there other things to include or link out to?

**Comment by consideRatio** [comment]
Ping to @choldgraf @willingc who I think undertands a lot of these details - if you have time I'd be very thankful for a review my understanding above!

**Comment by choldgraf** [comment]
oh gosh, I'm not gonna lie, I find the `autodoc` stuff very confusing haha. It will take some digging for me to re-re-re-remember how to get it working again :-)

**Comment by consideRatio** [comment]
Ah, I figure we should avoid investing two people's time to independently figuring this out. I mainly looked for a review to conclude "this doesn't contradict my understanding".

I've ended up thinking about related things for hours now and feel more confident this is a reasonable understanding.
